patches/minecraft/net/minecraft/server/network/ServerGamePacketListenerImpl.java.patch

```diff
@@ -53,7 +53,16 @@
              }, ChatType.CHAT, this.f_9743_.m_142081_());
           }
  
-@@ -1222,6 +_,7 @@
+@@ -1216,12 +_,14 @@
+             return;
+          }
+ 
+-         double d0 = 36.0D;
+-         if (this.f_9743_.m_20280_(entity) < 36.0D) {
++         double maxRange = this.f_9743_.m_21133_(net.minecraftforge.common.ForgeMod.ATTACK_RANGE.get()) + 3; //Forge: Change constant value to max possible range. Equal to current value + 3 (because creative gives +3)
++         maxRange = Math.max(maxRange, this.f_9743_.m_21133_(net.minecraftforge.common.ForgeMod.REACH_DISTANCE.get()) + 0.5); //Used for both attacks and interacts, needs to be the highest of the two for sanity.
++         if (this.f_9743_.m_20280_(entity) < maxRange * maxRange) {
```

Comment:
This solution won't work with entities that have a very large bounding box and/or players with reduced attack range.
The method distanceSqTo uses the **position** of both entities to calculate the distance which is problematic when the bounding box of the victim keeps the player out of reach of being able to make a attack that is "visually" in range.

For testing purposes you can reduce the attack range and summon a big slime (e.g. `/summon minecraft:slime ~ ~ ~ {Size: 15}`)
(note: don't forget to reduce the block reach as well -> see line 63)

Reply:
![distanceIssue](https://user-images.githubusercontent.com/6061273/156928452-76f79fb4-cb96-41e9-85ae-0947c4063372.png)

Reply:
Note that this is what vanilla does in the first place - so this issue (with arbitrarily large entities) exists there as well.  I'm not sure if it is documented on Mojang's end.

The only possible fix would be to create a vector from the player's eye position to somewhere on the entity (maybe the center) and ask the AABB for an intersection, then use the distance between that intersection point and the player's eye position.

I'm not sure what other problems that might lead to, however...

Reply:
That's what I've done for a while now, and it seems fine.
https://github.com/Jackiecrazy/WarDance/blob/51a0da44570073dfb683e6e925d139c74c8dd0a1/src/main/java/jackiecrazy/wardance/utils/GeneralUtils.java#L112-L123

Reply:
i could also imagine that vanilla accepts all attacks within 6 blocks in order to mitigate issues related to network lag. But who knows for sure, could just be legacy code.

Reply:
> I'm not sure if it is documented on Mojang's end.

I couldn't find anything specific in the minecraft bug tracker related to attacking large bounding boxes.

Reply:
> The only possible fix would be to create a vector from the player's eye position to somewhere on the entity (maybe the center) and ask the AABB for an intersection, then use the distance between that intersection point and the player's eye position.

I did something like this. Haven't heard any complaints yet.
https://github.com/Elenterius/Combat-Commons/blob/1.18.x/src/main/java/com/github/elenterius/combat_commons/utils/RayTraceUtil.java#L57-L75

Reply:
Resolving as outdated - see https://github.com/MinecraftForge/MinecraftForge/pull/8478/files#diff-fad76e2aed5c02fd9756e9dda64030fb777b549c5fcbdfc56466af63a07efee6R67